src/modules/product/repositories/productRepositories.ts

```diff
@@ -26,4 +26,24 @@ const findByModelsAndAttributes = async (model: any, keyOne: string, keyTwo: str
     });
 }
 
-export default { createProduct, createShop, findShopByAttributes,findByModelsAndAttributes};
\ No newline at end of file
+const updateProductByAttributes = async (updatedKey:string, updatedValue:any, whereKey:string, whereValue:any) =>{
+    await Products.update({ [updatedKey]: updatedValue }, { where: { [whereKey]: whereValue} });
+    return await findShopByAttributes(Products ,whereKey, whereValue);
+ }
+
+const markExpiredProducts = async () => {

```

Comment:
This function should take parameter to work properly.

Reply:
this function checks all products and when the expiryDate of the product has passed it changes the status to unavailable and expired to true.